### PR TITLE
Use JSON only editor when editing a pod

### PR DESCRIPTION
**Pull request by rcorral**
refs: DCOS-11793

gif

### CODE REVIEW

**Comment by MatApple** [comment]
retest this please

**Comment by nLight** [comment]
After rebasing this is what I see, only couple of rows. But the logic works perfectly

screen shot 2016-12-01 at 10 17 35

**Comment by nLight** [comment]
I think this visual bug is unrelated, so you have my label

**Comment by nLight** [comment]
I merged the fix and JSON editor is fullscreen again \o/

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js [hunk 1]
```diff
@@ -69,9 +69,21 @@ class NewServiceFormModal extends Component {
...
-      });
+      let newState = {serviceConfig: nextProps.service.getSpec()};
+
+      if (nextProps.isEdit) {
+        newState.servicePickerActive = false;
+
+        if (nextProps.service instanceof Pod) {
+          newState.serviceJsonActive = true;
+          newState.serviceFormActive = false;
+        } else {
```
Comment (orlandohohmeier) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Let's remove the `else` case by moving the respective lines to the top.

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js [hunk 2]
```diff
@@ -98,6 +104,13 @@ class NewServiceFormModal extends Component {
       return;
     }
 
+    // Close if picker is open, or if editing a service in the form
+    if (servicePickerActive ||
```
Comment (orlandohohmeier) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Any chance that we can simplify this condition?

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js [hunk 3]
```diff
@@ -371,16 +384,28 @@ class NewServiceFormModal extends Component {
     // Switch directly to form if edit
     if (nextProps.isEdit) {
       newState.servicePickerActive = false;
-      newState.serviceFormActive = true;
+
+      if (nextProps.service instanceof Pod) {
+        newState.serviceJsonActive = true;
+      } else {
+        newState.serviceFormActive = true;
```
Comment (orlandohohmeier) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Let's remove the `else` case here as well

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js [hunk 4]
```diff
@@ -371,16 +384,28 @@ class NewServiceFormModal extends Component {
...
-    let {servicePickerActive, serviceFormActive} = this.state;
+    let {
+      serviceFormActive,
+      servicePickerActive,
+      serviceReviewActive,
+      serviceJsonActive
+    } = this.state;
     let label = 'Back';
-    if (servicePickerActive || (serviceFormActive && this.props.isEdit)) {
+    if (servicePickerActive ||
```
Comment (orlandohohmeier) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Please simplify this condition or document the different cases

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js [hunk 5]
```diff
@@ -98,6 +104,13 @@ class NewServiceFormModal extends Component {
       return;
     }
 
+    // Close if picker is open, or if editing a service in the form
+    if (servicePickerActive ||
+      ((serviceFormActive || serviceJsonActive) && this.props.isEdit)) {
+      this.handleClose();
+      return;
```
Comment (MatApple) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: New line

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js (hunk 4 above)
Comment (MatApple) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Let's also add a newline above here - this block of `let`s and `if` is hard to parse.

File: plugins/services/src/js/components/modals/NewCreateServiceModal.js (hunk 5 above)
Comment (orlandohohmeier) [inline plugins/services/src/js/components/modals/NewCreateServiceModal.js]: Thought we don't do a newline in this particular case...

**Comment by orlandohohmeier** [comment]
@MatApple addressed all comments...
